### PR TITLE
[Feature] Multiversion cache

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/exec/pipeline/pipeline_driver.cpp [hunk 1]
```diff
@@ -227,6 +227,9 @@ StatusOr<DriverState> PipelineDriver::process(RuntimeState* runtime_state, int w
                         {
                             SCOPED_TIMER(next_op->_push_timer);
                             QUERY_TRACE_SCOPED(next_op->get_name(), "push_chunk");
+                            LOG(INFO) << strings::Substitute("Transfer chunk: from=$0, to=$1, rows=$2, driver=$3",
```
Comment (kangkaisen) [inline be/src/exec/pipeline/pipeline_driver.cpp:230]: debug log?

File: be/src/exec/pipeline/scan/morsel.h [hunk 2]
```diff
@@ -63,8 +63,12 @@ class Morsel {
         return std::tuple<int64_t, int64_t>{0L, 0L};
     }
 
+    void set_from_version(int64_t from_version) { _from_version = from_version; }
+    int64_t from_version() { return _from_version; }
```
Comment (kangkaisen) [inline be/src/exec/pipeline/scan/morsel.h:67]: why don't need to version?

File: be/src/exec/vectorized/aggregator.cpp [hunk 3]
```diff
@@ -299,7 +300,20 @@ Status Aggregator::prepare(RuntimeState* state, ObjectPool* pool, RuntimeProfile
...
-Status Aggregator::reset_state(std::vector<vectorized::ChunkPtr>&& chunks) {
+Status Aggregator::reset_state(starrocks::RuntimeState* state, const std::vector<vectorized::ChunkPtr>& chunks,
+                               pipeline::Operator* op) {
+    RETURN_IF_ERROR(_reset_state(state));
+    begin_pending_reset_state();
+    for (const auto& chunk : chunks) {
+        if (chunk == nullptr || chunk->is_empty()) {
+            continue;
+        }
+        RETURN_IF_ERROR(op->push_chunk(state, chunk));
```
Comment (kangkaisen) [inline be/src/exec/vectorized/aggregator.cpp:311]: what's  op?

File: be/src/exec/vectorized/aggregator.cpp [hunk 4]
```diff
@@ -299,7 +300,20 @@ Status Aggregator::prepare(RuntimeState* state, ObjectPool* pool, RuntimeProfile
     return Status::OK();
 }
 
-Status Aggregator::reset_state(std::vector<vectorized::ChunkPtr>&& chunks) {
+Status Aggregator::reset_state(starrocks::RuntimeState* state, const std::vector<vectorized::ChunkPtr>& chunks,
+                               pipeline::Operator* op) {
+    RETURN_IF_ERROR(_reset_state(state));
+    begin_pending_reset_state();
```
Comment (kangkaisen) [inline be/src/exec/vectorized/aggregator.cpp:306]: during begin and end pending state, do we need the lock?

File: be/src/exec/vectorized/aggregator.cpp (hunk 3 above)
Comment (satanson) [reply be/src/exec/vectorized/aggregator.cpp:311]: comment added to the method proto

File: be/src/exec/vectorized/aggregator.cpp (hunk 4 above)
Comment (satanson) [reply be/src/exec/vectorized/aggregator.cpp:306]: no, it just tells aggregator, the refill-back chunks are intermediate type, so aggregator should invoke merge method of agg func to process these chunks. there is no concurrent issue.

**Review by kangkaisen — DISMISSED** [review]

**Comment by satanson** [comment]
run starrocks_clang-format

**Review by stdpain — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**